pkg/hintrunner/zero/zerohint_signature.go

```diff
@@ -0,0 +1,87 @@
+package zero
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	secp_utils "github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/utils"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	"github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+)
+
+func newVerifyZeroHint(val, q hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "VerifyZero",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//> q, r = divmod(pack(ids.val, PRIME), SECP_P)
+			//> assert r == 0, f"verify_zero: Invalid input {ids.val.d0, ids.val.d1, ids.val.d2}."
+			//> ids.q = q % PRIME
+
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			secPBig, ok := secp_utils.GetSecPBig()
+			if !ok {
+				return fmt.Errorf("GetSecPBig failed")
+			}
+
+			valAddr, err := val.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			valMemoryValues, err := hinter.GetConsecutiveValues(vm, valAddr, int16(3))
+			if err != nil {
+				return err
+			}
+
+			// [d0, d1, d2]
+			var valValues [3]*fp.Element
+
+			for i := 0; i < 3; i++ {
+				valValue, err := valMemoryValues[i].FieldElement()
+				if err != nil {
+					return err
+				}
+				valValues[i] = valValue
+			}
```

Comment:
I've seen this is used in several places in the code for different hints. I think it would be beneficial to encapsulate this logic in a separate method returning the  [3]*fp.Element structure. Maybe something like `ResolveAsBigInt3`

Reply:
True. I'll add it.

Reply:
Can I add this in a separate PR? Many hints(both in main and other PRs) will need to be modified.

Reply:
yes, I've created issue #346 for this. let's solve it in a separate PR